utilities/imgls.pl

```diff
@@ -0,0 +1,562 @@
+    $opts{'s'} and $opts{'show_blocks'}++;
+}
+
+sub get_dir_content {
+    my ($path, $filter, $filesref, $dirsref) = @_;
+    my $dh;
+
+    unless (opendir($dh, $path)) {
+        say "Unable to open directory $path: $!";
+        return undef;
+    }
+
+    while (readdir($dh)) {
+        next if defined $filter and $_ =~ /$filter/;
+        my $p = "$path/$_";
+        if (-d _lstat($p)) {
+            push @$dirsref, $p;
+        }
+        else {
+            push @$filesref, $p;
+        }
+    }
+    closedir $dh;
+    return 1;
+}
+
+sub do_ls {
+    my $path = shift;
+
+    my $blocks_total = 0;
+    my (@hfiles, %widths, $st);
+    for my $file (ls_sort(@_)) {
+        #say "FILE: $file";
+
+        my %h;
+        $h{'file'}      = $file;
+        $h{'filename'}  = defined $path ? (split /\//, $file)[-1] : $file;
+        $h{'st'}        = $st = _lstat($file);
+        $h{'ino'}       = $st->ino                      if $opts{'i'};
+        $h{'bytes'}     = $st->size;
+        $h{'bytesh'}    = format_human($h{'bytes'})     if $opts{'h'};
+        $h{'dims'}      = get_dimensions($file)         if $opts{'dimensions'} and -f $st && -r $st && $h{'bytes'};
+        $h{'nlink'}     = $st->nlink                    if $opts{'s'} or $opts{'l'};
+
+        if ($opts{'show_blocks'} or $opts{'l'}) {
+            $h{'blocks'} = $st->blocks;
+            if ( ! -d $st and ($opts{'a'} or $h{'filename'} !~ /^\.[^.]+/)) {
+                $blocks_total += $st->blocks;
+            }
+        }
+
+        if ($opts{'l'}) {
+            $h{'lsmodes'} = Stat::lsMode::format_mode($st->mode);
+            $h{'owner'} = ($opts{'n'} ? $st->uid : getpwuid $st->uid) // $st->uid;
+            $h{'group'} = ($opts{'n'} ? $st->gid : getgrgid $st->gid) // $st->gid       if not $opts{'o'};
+            $h{'time'}  = format_time($opts{'c'} ? $st->ctime : $st->mtime);
+        }
+        push @hfiles, \%h;
+
+        $widths{'dim_w'}  = max(defined $h{'dims'} ? length($h{'dims'}{'width'})  : 0, $widths{'dim_w'} // 0);
+        $widths{'dim_h'}  = max(defined $h{'dims'} ? length($h{'dims'}{'height'}) : 0, $widths{'dim_h'} // 0);
+        for my $key (qw/blocks ino bytes bytesh owner group nlink/) {
+            $widths{$key} = max(length($h{$key}), $widths{$key} // 0)   if exists $h{$key};
+        }
+    }
+
+    # Header output when @ARGV was > 1, or after second dir
+    print "\n"                                          if $path and $do_newline;
+    print "$path:\n"                                    if $path and $do_header++;
+
+    #   total blocks inline when -d, as header when -l or -s
+    say "total ", $blocks_total / ($opts{'k'} ? 2 : 1)   if $path and ! $opts{'d'} and ($opts{'show_blocks'} or $opts{'l'});
+
+    for my $h (@hfiles) {
+        if (! -f $h->{'st'} or ! $h->{'bytes'} or ($opts{'dimensions'} and ! $h->{'dims'} and ! $opts{'unknown'})) {
+            # may not produce correct num of dashes: --pre --h 3 --w 10
```

Comment:
Remove these two lines

Reply:
Oops, the first two comments don't apply anymore.  Done.